app/src/main/java/com/danielkim/soundrecorder/fragments/RecordFragment.java

```diff
@@ -151,29 +175,34 @@ public void onChronometerTick(Chronometer chronometer) {
         } else {
             //stop recording
             mRecordButton.setImageResource(R.drawable.ic_mic_white_36dp);
-            //mPauseButton.setVisibility(View.GONE);
+            mPauseButton.setVisibility(View.GONE);
             mChronometer.stop();
             mChronometer.setBase(SystemClock.elapsedRealtime());
             timeWhenPaused = 0;
             mRecordingPrompt.setText(getString(R.string.record_prompt));
 
+            //handle case : user press stop after pause
+            if(!mPauseRecording) mPauseRecording = true;
             getActivity().stopService(intent);
+            getActivity().unbindService(mConnection);
+
             //allow the screen to turn off again once recording is finished
             getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
         }
     }
 
-    //TODO: implement pause recording
     private void onPauseRecord(boolean pause) {
         if (pause) {
             //pause recording
+            mRecordingService.pauseRecording();
             mPauseButton.setCompoundDrawablesWithIntrinsicBounds
                     (R.drawable.ic_media_play ,0 ,0 ,0);
             mRecordingPrompt.setText((String)getString(R.string.resume_recording_button).toUpperCase());
```

Comment:
bug: change `mPauseButton` instead of `mRecordingPrompt`. set `mRecordingPrompt` text to "Paused".